scripts/asr_language_modeling/ngram_lm/opengrm.py

```diff
@@ -0,0 +1,285 @@
+alpha * ngram_a + beta * ngram_b = 2 * ngram_a + 1 * ngram_b
+
+python3 opengrm.py  --kenlm_bin_path /workspace/nemo/decoders/kenlm/build/bin/build_binary \
+                    --arpa_a /path/ngram_a.kenlm.tmp.arpa \
+                    --alpha 2 \
+                    --arpa_b /path/ngram_b.kenlm.tmp.arpa \
+                    --beta 1 \
+                    --out_path /path/out \
+
+
+Merge two N-gram language models and calculate its perplexity with test_file.
+python3 opengrm.py  --kenlm_bin_path /workspace/nemo/decoders/kenlm/build/bin/build_binary \
+                    --arpa_a /path/ngram_a.kenlm.tmp.arpa \
+                    --alpha 0.5 \
+                    --arpa_b /path/ngram_b.kenlm.tmp.arpa \
+                    --beta 0.5 \
+                    --out_path /path/out \
+                    --tokenizer_model_file /path/to/model_tokenizer.nemo \
+                    --test_file /path/to/test_manifest.json \
+                    --force
+"""
+
+import argparse
+import os
+import subprocess
+import sys
+
+import torch
+
+import nemo.collections.asr as nemo_asr
+from nemo.collections.asr.parts.submodules.ctc_beam_decoding import DEFAULT_TOKEN_OFFSET
+from nemo.collections.common.tokenizers.sentencepiece_tokenizer import SentencePieceTokenizer
+from nemo.utils import logging
+
+
+def ngrammerge(arpa_a, alpha, arpa_b, beta, arpa_c, force):
+    mod_a = arpa_a + ".mod"
+    mod_b = arpa_b + ".mod"
+    mod_c = arpa_c + ".mod"
+    if os.path.isfile(mod_c) and not force:
+        logging.info("File " + mod_c + " exists. Skipping.")
+    else:
+        sh_args = [
+            "ngrammerge",
+            "--alpha=" + str(alpha),
+            "--beta=" + str(beta),
+            "--normalize",
+            # "--use_smoothing",
+            mod_a,
+            mod_b,
+            mod_c,
+        ]
+        logging.info(
+            "\n"
+            + str(subprocess.run(sh_args, capture_output=False, text=True, stdout=sys.stdout, stderr=sys.stderr,))
+            + "\n",
+        )
+    return mod_c
+
+
+def arpa2mod(arpa_path, force):
+    mod_path = arpa_path + ".mod"
+    if os.path.isfile(mod_path) and not force:
+        return "File " + mod_path + " exists. Skipping."
+    else:
+        sh_args = [
+            "ngramread",
+            "--ARPA",
+            arpa_path,
+            mod_path,
+        ]
+        return subprocess.run(sh_args, capture_output=False, text=True, stdout=sys.stdout, stderr=sys.stderr,)
+
+
+def merge(arpa_a, alpha, arpa_b, beta, out_path, force):
+    logging.info("\n" + str(arpa2mod(arpa_a, force)) + "\n")
+
+    logging.info("\n" + str(arpa2mod(arpa_b, force)) + "\n")
+    arpa_c = os.path.join(out_path, f"{os.path.split(arpa_a)[1]}-{alpha}-{os.path.split(arpa_b)[1]}-{beta}.arpa",)
+    mod_c = ngrammerge(arpa_a, alpha, arpa_b, beta, arpa_c, force)
+    return mod_c, arpa_c
+
+
+def make_symbol_list(tokenizer_model_file, symbols, force):
```

Comment:
Being duplicated with above - maybe consider moving it out to Nemo ASR parts in new urils file